### PR TITLE
OOC 4354 ensure most recent date is not before first date

### PR DESCRIPTION
**Pull request by kathryn-dale**
Ticket: https://jira.collab.test-and-trace.nhs.uk/browse/OOC-4354

### CODE REVIEW

File: runner/src/server/forms/ReportAnOutbreak.json [hunk 1]
```diff
@@ -542,7 +542,9 @@
               "date.base": "The date that symptoms started must be a real date",
               "date.max": "The date that symptoms started must be in the past"
             },
-            "errorLabel": "The date that symptoms started"
+            "errorLabel": "The date that symptoms started",
+            "first_date_to_compare": "S4Q6",
```
Comment (emilyjevans) [inline runner/src/server/forms/ReportAnOutbreak.json:546]: We need to do same for Sections 5 and 6

File: runner/src/server/plugins/engine/pageControllers/validationOptions.ts [hunk 2]
```diff
@@ -51,6 +51,7 @@ export const messages: ValidationOptions["messages"] = {
   "date.min": messageTemplate.dateMin,
   "date.max": messageTemplate.dateMax,
   "date.format": messageTemplate.dateFormat,
+  "date.chronological": messageTemplate.dateChronological,
```
Comment (emilyjevans) [inline runner/src/server/plugins/engine/pageControllers/validationOptions.ts:54]: where's the corresponding `dateChronological` value?

File: runner/src/server/plugins/engine/components/DatePartsField.ts [hunk 3]
```diff
@@ -123,15 +131,51 @@ export class DatePartsField extends FormComponent {
...
+    const mostRecentDateDay = payload[`${most_recent_date_name}__day`];
+    const mostRecentDateMonth = payload[`${most_recent_date_name}__month`];
+    const mostRecentDateYear = payload[`${most_recent_date_name}__year`];
+
+    if (day || month || year) {
       const indexedMonth = month - 1; // Adjust month for zero-based index
       const parsedDate = new Date(year, indexedMonth, day);
 
       if (month - 1 === parsedDate.getMonth()) {
+        if (name == most_recent_date_name) {
```
Comment (emilyjevans) [inline runner/src/server/plugins/engine/components/DatePartsField.ts:154]:
```suggestion
        if (name === most_recent_date_name) {
```

File: runner/src/server/forms/ReportAnOutbreak.json (hunk 1 above)
Comment (kathryn-dale) [reply runner/src/server/forms/ReportAnOutbreak.json:546]: Oh good catch, I was so excited it worked for the first section i completely forgot it was 1 of several that needed doing

File: runner/src/server/plugins/engine/pageControllers/validationOptions.ts (hunk 2 above)
Comment (kathryn-dale) [reply runner/src/server/plugins/engine/pageControllers/validationOptions.ts:54]: Oh, I don't remember adding that in. I'll have a look

Comment (emilyjevans) [reply runner/src/server/plugins/engine/pageControllers/validationOptions.ts:54]: Sorry I mean there's not a `dateChronological ` value in the `messageTemplate` object above in this file so I think this evaluates to null?

Comment (kathryn-dale) [reply runner/src/server/plugins/engine/pageControllers/validationOptions.ts:54]: @masuk-kazi98 Is this something you'e worked on? Do you know if this value is needed? I don't recognise it and it's not in upstream but it's used in the getCustomDateValidator function in helpers.ts

Comment (masuk-kazi98) [reply runner/src/server/plugins/engine/pageControllers/validationOptions.ts:54]: this was added when i first tested out an implementation for the date-against-date check. won't be needed if you've set your own message

**Review by emilyjevans — APPROVED** [review]

**Pull request merged**